### ISSUE TITLE
Not working properly with Glide

### ISSUE THREAD

**Issue by krishnameena** [google/flexbox-layout#204]
I am using Glide library to load images in RecyclerView but when I am scrolling the RecyclerView sometimes 2 images showing per row some time 1 image per row.
Following is my code 

`
FlexboxLayoutManager layoutManager = new FlexboxLayoutManager();
        layoutManager.setFlexWrap(FlexWrap.WRAP);
        layoutManager.setFlexDirection(FlexDirection.ROW);
        layoutManager.setAlignItems(AlignItems.STRETCH);
        mRecyclerView.setLayoutManager(layoutManager);
`

```
private void setImageView(final ImageView imageView, Integer resourceId) {
        Glide.with(mContext)
                .load(resourceId)
                .asBitmap()
                .format(DecodeFormat.PREFER_RGB_565)
                .diskCacheStrategy(DiskCacheStrategy.RESULT)
                //.skipMemoryCache(true)
                .thumbnail(0.1f)
                .placeholder(resourceId)
                .error(resourceId)
                .centerCrop()
                .listener(new RequestListener<Integer, Bitmap>() {
                    @Override
                    public boolean onException(Exception e, Integer model, Target<Bitmap> target, boolean isFirstResource) {
                        return false;
                    }

                    @Override
                    public boolean onResourceReady(Bitmap resource, Integer model, Target<Bitmap> target, boolean isFromMemoryCache, boolean isFirstResource) {
                        ViewGroup.LayoutParams lp = imageView.getLayoutParams();
                        if (lp instanceof FlexboxLayoutManager.LayoutParams) {
                            FlexboxLayoutManager.LayoutParams flexboxLp = (FlexboxLayoutManager.LayoutParams)
                                    imageView.getLayoutParams();
                            flexboxLp.setFlexGrow(1.0f);
                        }
                        return false;
                    }
                })
                .into(imageView);
    }
```

XML file 

`<?xml version="1.0" encoding="utf-8"?>
<ImageView xmlns:android="http://schemas.android.com/apk/res/android"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:id="@+id/image"
    android:src="@drawable/ic_guest_services_light"
    android:scaleType="centerCrop"
    android:layout_margin="1dp"
    android:contentDescription="@null"/>`

**Comment by thagikura** [comment]
Hi,

Thanks for reporting this.
Just to make sure, do your images have the same width or each one has different width?

I'm asking because it's likely to be an intended behavior if the width of each image is different.
Since depending on the width of an image, FlexboxLayoutManager determines if the image being measured should be on the next line or in the current line.
(I'm assuming you are setting the flexWrap property as FlexWrap.WRAP, it's the default value)

**Comment by krishnameena** [comment]
I have different different size images and I am setting the flexWrap property as FlexWrap.WRAP

**Comment by thagikura** [comment]
Ok so I think it's likely to be an intended behavior.

I guess the rows which have 1 image in it don't have enough space to include the second image, the rows which have 2 images were able to contain 2 images (but not 3 images).

Could you check if my assumption is correct?

**Comment by krishnameena** [comment]
Let me explain in detail let say in row number 5 there is 2 images in that row now when I scroll RecyclerView to bottom and again if I scroll down it to row 5 than the 2 images which were in 5th row, now they are appearing in 2 rows (row no 5 and row no 6 ) please see screenshots for details[
![screenshot_20170223-163821](https://cloud.githubusercontent.com/assets/10487399/23256864/4976085a-f9e7-11e6-9af5-5f4314d38bfa.png)
![screenshot_20170223-163843](https://cloud.githubusercontent.com/assets/10487399/23256865/49bb93ca-f9e7-11e6-8456-1b400ab516d8.png)

](url)

**Comment by thagikura** [comment]
Hi @krishnameena,

Apologies for this thread hasn't been updated long.
I just released a new version (0.3.0-alpha3) of the RecyclerView integration with bunch of bug fixes.
I suspect that may be related to this issue.

Could you try it and see it's still happening?

**Comment by wightwulf1944** [comment]
Hi @krishnameena 

I've tried reproducing this in the latest version and I regret to say that this bug is still happening.

FlexboxLayoutManager version: 0.3.0-alpha3
Glide version: 3.7.0

Repro steps:

1. Clone demo-cat-gallery
2. Replace `mImageView.setImageDrawable(drawable)` in CatViewHolder with Glide load like so
```
Glide.with(mImageView.getContext())
                .load(drawableResourceId)
                .into(mImageView);
```

**Result:** 
![screenshot_20170403-211301](https://cloud.githubusercontent.com/assets/2737347/24610887/d646afde-18b2-11e7-9c26-d297863aeeac.png)

**Expected result:**
![screenshot_20170403-211428](https://cloud.githubusercontent.com/assets/2737347/24610892/de89b678-18b2-11e7-8a01-0a5ca68b4d99.png)

**Comment by krishnameena** [comment]
Yes I also checked and the bug is still there

**Comment by wightwulf1944** [comment]
I think it may have something to do with how Glide does things
https://github.com/bumptech/glide/issues/510

**Comment by thagikura** [comment]
Thanks for the pointer. Looks like it's exactly related to that issue on how Glide handles wrap_content.

From LayoutManager side, there isn't (and shouldn't do) any way of change its behavior.
But setting the fixed size worked for like other options (without Glide or with Picasso) me like

```java
Glide.with(mImageView.getContext())
        .load(drawableResourceId)
        .override(drawable.getIntrinsicWidth(), drawable.getIntrinsicHeight())
        .into(mImageView);
```

**Comment by krishnameena** [comment]
Thanks it is working like a charm now.

**Comment by wightwulf1944** [comment]
There are some issues with this approach when using Glide with a placeholder
```
Glide.with(context)
    .load(imageUrl)
    .placeholder(R.drawable.placeholder)
    .override(width, height)
    .into(imageView);
```

at first glide resizes the placeholder to fit into the current dimensions of the ImageView, then once the image is ready, Glide resizes the ImageView as specified in the override statement. This results in weird scrolling behavior where things jump around as they switch from one size to another as they load.

To fix this, I set the dimensions of the ImageView in onBindViewHolder along with loading with Glide like so:

```
ViewGroup.LayoutParams lp = imageView.getLayoutParams();
lp.width = width;
lp.height = height;
imageView.setLayoutParams(lp);

Glide.with(context)
    .load(imageUrl)
    .placeholder(R.drawable.placeholder)
    .into(imageView);
```

But then the imageView no longer grows or shrinks like a flexbox item when I do this. I've tried setting setFlexGrow, setMin or Max Height and Width, and all of them seems to be ignored once a definite height and width is set. Is this intended? How can I fix this?
